src/zenml/integrations/evidently/steps/evidently_profile.py

```diff
@@ -0,0 +1,112 @@
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+from typing import Optional, cast
+
+from evidently.model_profile import Profile  # type: ignore
+from evidently.pipeline.column_mapping import ColumnMapping  # type: ignore
+from evidently.profile_sections import (  # type: ignore
+    CatTargetDriftProfileSection,
+    ClassificationPerformanceProfileSection,
+    DataDriftProfileSection,
+    NumTargetDriftProfileSection,
+    ProbClassificationPerformanceProfileSection,
+    RegressionPerformanceProfileSection,
+)
+from evidently.profile_sections.base_profile_section import (  # type: ignore
+    ProfileSection,
+)
+
+from zenml.artifacts import DataArtifact
+from zenml.steps import StepContext
+from zenml.steps.step_interfaces.base_drift_detection_step import (
+    BaseDriftDetectionConfig,
+    BaseDriftDetectionStep,
+)
+
+profile_mapper = {
+    "datadrift": DataDriftProfileSection,
+    "categoricaltargetdrift": CatTargetDriftProfileSection,
+    "numericaltargetdrift": NumTargetDriftProfileSection,
+    "classificationmodelperformance": ClassificationPerformanceProfileSection,
+    "regressionmodelperformance": RegressionPerformanceProfileSection,
+    "probabilisticmodelperformance": ProbClassificationPerformanceProfileSection,
+}
+
+
+class EvidentlyProfileConfig(BaseDriftDetectionConfig):
+    """Config class for Evidently profile steps.
+
+    column_mapping: properties of the dataframe's columns used
+    profile_section: a string that identifies the profile section to be used.
+        The following are valid options supported by Evidently:
+        - "datadrift"
+        - "categoricaltargetdrift"
+        - "numericaltargetdrift"
+        - "classificationmodelperformance"
+        - "regressionmodelperformance"
+        - "probabilisticmodelperformance"
+    """
+
+    def get_profile_section(self) -> ProfileSection:
+        try:
+            return profile_mapper[self.profile_section]()
+        except KeyError:
+            raise ValueError(
+                f"Invalid profile section: {self.profile_section}"
+                "\n\n"
+                "Valid and supported options are: \n"
+                "- datadrift, \n"
+                "- categoricaltargetdrift, \n"
+                "- numericaltargetdrift, \n"
+                "- classificationmodelperformance, \n"
+                "- regressionmodelperformance, \n"
+                "- probabilisticmodelperformance \n"
```

Comment:
I like how descriptive this error message it. You should use `profile_mapper.keys()` to populate this list and save yourself the trouble of having to keep them in sync.

Reply:
I had initially tried to do that, but couldn't figure out how to add those individual items as arguments for the `KeyError` error raising. I'll try again. I agree that it would be better.